src/test/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorDataSourceDownloaderIntegrationTest.java

```diff
@@ -0,0 +1,134 @@
+                        false,
+                        false
+                },
+                {
+                        true,
+                        true,
+                        true
+                },
+                {
+                        true,
+                        false,
+                        true
+                },
+        };
+    }
+
+    @DataProvider
+    private Object[][] provideForTestDownload() {
+        return new Object[][] {
+                {
+                        FuncotatorDataSourceDownloader.SOMATIC_ARG_LONG_NAME,
+                        true,
+                        true,
+                        false
+                },
+                {
+                        FuncotatorDataSourceDownloader.SOMATIC_ARG_LONG_NAME,
+                        true,
+                        false,
+                        false
+                },
+                {
+                        FuncotatorDataSourceDownloader.GERMLINE_ARG_LONG_NAME,
+                        true,
+                        true,
+                        false
+                },
+                {
+                        FuncotatorDataSourceDownloader.GERMLINE_ARG_LONG_NAME,
+                        true,
+                        false,
+                        false
+                },
+                {
+                        FuncotatorDataSourceDownloader.SOMATIC_ARG_LONG_NAME,
+                        true,
+                        false,
+                        true
+                }
+        };
+    }
+
+    //==================================================================================================================
+    // Tests:
+
+    @Test(enabled = doDebugTests,
+            dataProvider = "provideForTestDownload",
+            groups = {"funcotatorValidation", "bucket"}
+    )
+    void testDownloadRealDataSources(final String dsTypeArg, final boolean doOverwrite, final boolean doValidate, final boolean doExtract) {
+        final ArgumentsBuilder arguments = new ArgumentsBuilder();
+
+        arguments.addBooleanArgument(dsTypeArg, true);
+        arguments.addBooleanArgument(FuncotatorDataSourceDownloader.OVERWRITE_ARG_LONG_NAME, doOverwrite);
+        arguments.addBooleanArgument(FuncotatorDataSourceDownloader.VALIDATE_INTEGRITY_ARG_LONG_NAME, doValidate);
+        arguments.addBooleanArgument(FuncotatorDataSourceDownloader.EXTRACT_AFTER_DOWNLOAD, doExtract);
+        arguments.addArgument("verbosity", "INFO");
+
+        runCommandLine(arguments);
+    }
+
+    @Test(dataProvider = "provideForTestDownloadSmallDummyDataSources",
+            groups = {"bucket"})
+    void testDownloadDummySmallDataSources(final boolean doOverwrite, final boolean doValidate, final boolean doExtract) {
+
+        // Create an output location for the data sources to go:
+        final File tmpDir = createTempDir("FuncotatorDataSourceDownloaderIntegrationTest_testDownloadDummySmallDataSources");
+        tmpDir.deleteOnExit();
```

Comment:
`createTempDir()` calls `deleteRecursivelyOnExit()` for you

Reply:
Fixed.